### PR TITLE
Fixed Changing sample for downloadable product failure

### PR DESCRIPTION
**Pull request by ravi-chandra3197**
### Description (*)

Fixed Changing sample for downloadable product failure
### Fixed Issues (if relevant)

1. magento/magento2#6272: Changing sample for downloadable product failure

### Manual testing scenarios (*)

1. N/A

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @ravi-chandra3197. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

File: app/code/Magento/Downloadable/Model/SampleRepository.php [hunk 1]
```diff
@@ -308,8 +308,11 @@ protected function updateSample(
             $existingSample->setTitle($sample->getTitle());
         }
 
-        if ($sample->getSampleType() === 'file' && $sample->getSampleFileContent() === null) {
-            $sample->setSampleFile($existingSample->getSampleFile());
+        if ($sample->getSampleType() === 'file'
+            && $sample->getSampleFileContent() === null
+            && $sample->getSampleFileContent() !== null
```
Comment (orlangur) [inline app/code/Magento/Downloadable/Model/SampleRepository.php]: Such condition will always evaluate to false.

Did you mean `$sample->getSampleFile() !== null`?

**Review by orlangur — CHANGES_REQUESTED** [review]

File: app/code/Magento/Downloadable/Model/SampleRepository.php (hunk 1 above)
Comment (ravi-chandra3197) [reply app/code/Magento/Downloadable/Model/SampleRepository.php]: Hello @orlangur 
I have updated my code can review it.

**Comment by ravi-chandra3197** [comment]
Hello @orlangur
I have updated my code can review it.

**Comment by ravi-chandra3197** [comment]
Hello @orlangur
I have updated my code can review it.

**Review by orlangur — CHANGES_REQUESTED** [review]
Please squash changes into a single commit so that we have perfectly clean history 😉

**Comment by ravi-chandra3197** [comment]
Hello @orlangur
I have squash changes into a single commit.

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-4684](https://jira.corp.magento.com/browse/ENGCOM-4684) has been created to process this Pull Request

**Comment by VasylShvorak** [comment]
:heavy_check_mark: QA passed

**Pull request merged**